pupil_src/shared_modules/head_pose_tracker/worker/detection_worker.py

```diff
@@ -88,26 +88,32 @@ def offline_detection(
     yield None
 
     src = video_capture.File_Source(
-        SimpleNamespace(), source_path, fill_gaps=True, timing=None
+        SimpleNamespace(), source_path, fill_gaps=False, timing=None
     )
+    timestamps_no_gaps = src.timestamps
+    uncalculated_timestamps = all_timestamps[frame_indices]
+    seek_poses = np.searchsorted(timestamps_no_gaps, uncalculated_timestamps)
 
     queue = []
-    for frame_index in frame_indices:
-        shared_memory.progress = (frame_index - frame_start + 1) / frame_count
-        timestamp = timestamps[frame_index]
-        src.seek_to_frame(frame_index)
-        frame = src.get_frame()
+    for frame_index, timestamp, target_frame_idx in zip(
+        frame_indices, uncalculated_timestamps, seek_poses
+    ):
+        detections = []
+        if timestamp in timestamps_no_gaps:
+            src.seek_to_frame(target_frame_idx)
+            frame = src.get_frame()
+            detections = _detect(frame)
 
-        detections = _detect(frame)
         if detections:
-            serialized_dicts = [
-                fm.Serialized_Dict(detection) for detection in detections
-            ]
-            queue.append((timestamp, serialized_dicts, frame_index, len(detections)))
+            serialized_dicts = [fm.Serialized_Dict(d) for d in detections]
         else:
-            queue.append((timestamp, [fm.Serialized_Dict({})], frame_index, 0))
+            serialized_dicts = [fm.Serialized_Dict({})]
```

Comment:
```suggestion
            serialized_dicts = []
```
Shouldn't this be compatible?

Reply:
It was for saving the number of markers in a frame in pldata.topics.
Now I save `frame_index_to_num_markers` explicitly, so `serialized_dicts` can be `[]` when there's no marker in the frame.